### PR TITLE
use ~/.cargo if CARGO_HOME is unset

### LINKED ISSUE #42: feature request: assume CARGO_HOME=~/.cargo if environment variable is unset?
Trouble I'm having:
```
[rwalker@ruffian solana]$ cargo audit 
thread 'main' panicked at 'Can't locate CARGO_HOME!', /home/rwalker/.cargo/registry/src/github.com-1ecc6299db9ec823/rustsec-0.9.1/src/repository/mod.rs:54:13
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```
Works fine with CARGO_HOME set, but I normally don't have it set.
```
[rwalker@ruffian solana]$ CARGO_HOME=~/.cargo cargo audit 
    Fetching advisory database from `https://github.com/RustSec/advisory-db.git`
      Loaded 11 security advisories (from /home/rwalker/.cargo/advisory-db)
    Scanning Cargo.lock for vulnerabilities (197 crate dependencies)
error: Vulnerable crates found!

ID:      RUSTSEC-2018-0003
Crate:   smallvec
Version: 0.2.1
Date:    2018-07-19
URL:     https://github.com/servo/rust-smallvec/issues/96
Title:   Possible double free during unwinding in SmallVec::insert_many
Solution: upgrade to: >= 0.6.3 OR ^0.3.4 OR ^0.4.5 OR ^0.5.1

error: 1 vulnerability found!
[rwalker@ruffian solana]$ cargo audit --version
cargo-audit 0.5.2
[rwalker@ruffian solana]$ cargo --version
cargo 1.28.0 (96a2c7d16 2018-07-13)
[rwalker@ruffian solana]$ rustc --version
rustc 1.28.0 (9634041f0 2018-07-30)
```

### PR DESCRIPTION
**Pull request by rob-solana**
fixes #42

### CODE REVIEW

File: src/repository/mod.rs [hunk 1]
```diff
@@ -50,6 +50,10 @@ impl Repository {
     pub fn default_path() -> PathBuf {
         if let Some(path) = env::var_os("CARGO_HOME") {
             PathBuf::from(path).join(ADVISORY_DB_DIRECTORY)
+        } else if let Some(path) = env::var_os("HOME") {
```
Comment (tarcieri) [inline src/repository/mod.rs]: There's `std::env::home_dir()`, but it's deprecated, see:

https://internals.rust-lang.org/t/deprecate-or-break-fix-std-env-home-dir/7315/18

This seems a bit tricky. I think perhaps this is the best option?

https://docs.rs/directories/latest/directories/struct.UserDirs.html#method.home_dir

**Comment by tarcieri** [comment]
Looks good, thanks!

**Comment by rob-solana** [comment]
my pleasure.  how long until audit can consume an updated version of the crate?

**Comment by tarcieri** [comment]
I'd like to also address https://github.com/RustSec/cargo-audit/issues/32 before shipping another version of the crate

**Pull request merged**